01-Authentication-RS256/app/lib/auth0_client.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'jwt'
+require 'net/http'
+
+class Auth0Client 
+
+  # Auth0 Client Objects 
+  Error = Struct.new(:message, :status)
+  Response = Struct.new(:decoded_token, :error)
+  Token = Struct.new(:token) do
+    def validate_permissions(permissions)
+      required_permissions = Set.new permissions
+      token_permissions = Set.new token[0]['permissions']
```

Comment:
Should we use the token scopes as `token_permissions`?

E.g.:
```ruby
scopes = token[0]['scope']
token_permissions = scopes.present? ? Set.new(scopes.split(" ")) : Set.new
```

Reply:
I had a long conversation with @Sambego on this one 😄 seems like both permissions and scopes are correct but it depends on the case. I noticed most quickstarts use `scope` so I'll change it but I wonder if we should add a little explanation or link to why we are using `scope` and not `permissions`, not a conversation for this PR tho 😅

Reply:
Usually we use `scope` across QS, I think.